modules/contact/src/constraints/ComputeWeightedGapLMMechanicalContact.C

```diff
@@ -108,7 +107,12 @@ ComputeWeightedGapLMMechanicalContact::computeQpProperties()
     gap_vec(2).derivatives() =
         (*_primary_disp_z)[_qp].derivatives() - (*_secondary_disp_z)[_qp].derivatives();
 
-  _qp_gap = gap_vec * (_normals[_normal_index] * _JxW_msm[_qp] * _coord[_qp]);
+  if (_interpolate_normals)
+    _qp_gap = gap_vec * (_normals[_qp] * _JxW_msm[_qp] * _coord[_qp]);
+  else
+    _qp_gap_nodal = gap_vec * (_JxW_msm[_qp] * _coord[_qp]);
```

Comment:
It's tough sometimes when I don't write the code to have an intuitive grasp for everything. For instance I don't know whether it makes sense to have two different gap members?

Reply:
To use the design of these mortar contact constraint, I think we need to make some distinction between the quantities from  fully interpolated geometry and those resulting from the nodal geometry. Here, I am trying to have a nodal vector of interpolated quantities (`_qp_gap_nodal`)  which will be paired with a nodal normal vector in `computeQpIProperties` when `_interpolate_normals` is false, whereas the `_qp_gap` is a scalar from the fully interpolated strategy for the default `_interpolate_normals` true.

Reply:
that sounds fine. Just wanted to pick your brain about it